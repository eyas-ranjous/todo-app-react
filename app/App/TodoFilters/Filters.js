export default {
  All: 0,
  Todo: 1,
  Done: 2
};
