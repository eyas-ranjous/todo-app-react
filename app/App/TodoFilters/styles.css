.todo-filters {
  margin-bottom: 15px;
  width: 100%;
}

.todo-filters button {
  padding: 10px;
  border: none;
  min-width: 50px;
  cursor: pointer;
  background: #dddbdb;
  margin-bottom: 10px;
  font-size: 0.9em;
  border-radius: 20px;
  border: none;
  outline: none;
}

.todo-filters button:focus {
  outline: none;
}

.todo-filters .all {
  margin-right: 10px;
}

.todo-filters .all-selected {
  margin-right: 10px;
  background: #f2ce63;
}

.todo-filters .all:hover {
  background: #f2ce63;
}

.todo-filters .todo {
  margin-right: 10px;
}

.todo-filters .todo:hover {
  background: #f28282;
}

.todo-filters .todo-selected {
  margin-right: 10px;
  background: #f28282;
}

.todo-filters .done {
  margin-right: 10px;
}

.todo-filters .done:hover {
  background: #69db89;
}

.todo-filters .done-selected {
  margin-right: 10px;
  background: #69db89;
}
