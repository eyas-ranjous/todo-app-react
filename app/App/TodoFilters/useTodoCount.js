import { useState, useEffect } from 'react';

import TodoStatus from '../TodoStatus';

export default (todos) => {
  const [allCount, setAllCount] = useState(todos.size);
  const [todoCount, setTodoCount] = useState(todos.size);
  const [doneCount, setDoneCount] = useState(0);

  useEffect(() => {
    setAllCount(todos.size);
    const todoList = Array.from(todos).filter(
      ([, todo]) => todo.status === TodoStatus.todo
    );
    const doneList = Array.from(todos).filter(
      ([, todo]) => todo.status === TodoStatus.done
    );
    setTodoCount(todoList.length);
    setDoneCount(doneList.length);
  }, [todos]);

  return { allCount, todoCount, doneCount };
};
