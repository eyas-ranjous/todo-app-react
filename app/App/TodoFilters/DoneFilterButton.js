import React from 'react';
import { useSelector } from 'react-redux';
import classNames from 'classnames';
import styles from './styles.css';

const DoneFilterButton = ({ setFilter }) => {
  const TodoStatus = useSelector((state) => state.TodoStatus);
  const TodoFilters = useSelector((state) => state.TodoFilters);
  const selectedFilter = useSelector((state) => state.selectedFilter);

  const filter = TodoFilters.DONE;
  const statusList = [TodoStatus.DONE];

  const classes = classNames(
    styles['filter-button'],
    styles['filter-button-done'],
    selectedFilter === filter ? styles['filter-button-selected'] : ''
  );

  return (
    <button
      className={classes}
      onClick={() => setFilter(statusList, filter)}
    >
      Done
    </button>
  );
};

export default DoneFilterButton;
