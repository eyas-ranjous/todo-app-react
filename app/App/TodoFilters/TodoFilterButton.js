import React from 'react';
import { useSelector } from 'react-redux';
import classNames from 'classnames';
import styles from './styles.css';

const TodoFilterButton = ({ setFilter }) => {
  const TodoStatus = useSelector((state) => state.TodoStatus);
  const TodoFilters = useSelector((state) => state.TodoFilters);
  const selectedFilter = useSelector((state) => state.selectedFilter);

  const filter = TodoFilters.TODO;
  const statusList = [TodoStatus.TODO];

  const classes = classNames(
    styles['filter-button'],
    styles['filter-button-todo'],
    selectedFilter === filter ? styles['filter-button-selected'] : ''
  );

  return (
    <button className={classes} onClick={() => setFilter(statusList, filter)}>
      To Do
    </button>
  );
};

export default TodoFilterButton;
