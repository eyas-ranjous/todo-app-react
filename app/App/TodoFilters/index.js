import React from 'react';

import TodoStatus from '../TodoStatus';
import styles from './styles.css';

export default ({ todoFilter, toggleTodoFilter }) => (
  <div className={styles['todo-filters']}>
    <button
      className={styles[`todo${todoFilter.has(TodoStatus.todo) ? '-selected' : ''}`]}
      onClick={() => toggleTodoFilter(TodoStatus.todo)}
    >
      To Do
    </button>

    <button
      className={styles[`done${todoFilter.has(TodoStatus.done) ? '-selected' : ''}`]}
      onClick={() => toggleTodoFilter(TodoStatus.done)}
    >
      Done
    </button>
  </div>
);
