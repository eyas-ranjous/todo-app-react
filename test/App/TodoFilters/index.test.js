import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'
import createStore from '../../../app/store';
import TodoFilters from '../../../app/App/TodoFilters';

describe('TodoFilters', () => {
  const store = createStore();

  it('renders filter buttons', () => {
    const setFilter = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <TodoFilters />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
