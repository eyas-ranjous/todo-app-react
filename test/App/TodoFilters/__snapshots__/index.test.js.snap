// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilters renders filter buttons 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <TodoFilters>
    <div>
      <AllFilterButton
        setFilter={[Function]}
      >
        <button
          className="filter-button filter-button-all filter-button-selected"
          onClick={[Function]}
        >
          All
        </button>
      </AllFilterButton>
      <TodoFilterButton
        setFilter={[Function]}
      >
        <button
          className="filter-button filter-button-todo"
          onClick={[Function]}
        >
          To Do
        </button>
      </TodoFilterButton>
      <DoneFilterButton
        setFilter={[Function]}
      >
        <button
          className="filter-button filter-button-done"
          onClick={[Function]}
        >
          Done
        </button>
      </DoneFilterButton>
    </div>
  </TodoFilters>
</Provider>
`;
